### PR TITLE
fix node not belong to ASG issue when scaling down

### PR DESCRIPTION
**Pull request by patrickshan**
* error out the app when found node not inside an expected ASG
* move all log.Fatal cases out of pkg into main app
* change errors pkg consistent everywhere to `github.com/pkg/errors`

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -110,18 +110,23 @@ func setupNodeGroups() []controller.NodeGroupOptions {
...
 		if *leaderElect {
 			log.Warn("Doing leader election out of cluster is not recommended.")
 		}
-		k8sClient = k8s.NewOutOfClusterClient(*kubeConfigFile)
-	} else {
-		log.Info("Using in cluster config")
-		k8sClient = k8s.NewInClusterClient()
+		k8sClient, err := k8s.NewOutOfClusterClient(*kubeConfigFile)
+		if err != nil {
+			log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (patrickshan) [reply cmd/main.go]: we can do that, actually I'll do the same for `setupNodeGroups` as well so we have all main goroutine fatal logs inside main function.

File: pkg/controller/client.go [hunk 2]
```diff
@@ -44,7 +45,7 @@ func NewClient(k8sClient kubernetes.Interface, nodegroups []NodeGroupOptions, st
 
 	synced := k8s.WaitForSync(3, stopCache, podSync, nodeSync)
 	if !synced {
-		log.Fatalf("Attempted to wait for caches to be synced %d times. Exiting..", 3)
+		return nil, errors.New("attempted to wait for caches to be synced %d times. Exiting")
```
[the remark this answers is not in the document]
Comment (patrickshan) [reply pkg/controller/client.go]: missed that `%d` variable. I'll replace with the equivalent  from errors pkg `errors.Errorf()`.

File: cmd/main.go [hunk 3]
```diff
@@ -118,18 +116,18 @@ func setupK8SClient(kubeConfigFile *string, leaderElect *bool) kubernetes.Interf
...
 		if err != nil {
-			log.Fatal(err)
+			return nil, err
 		}
-		return k8sClient
+		return k8sClient, nil
 	}
 
 	log.Info("Using in cluster config")
 	k8sClient, err := k8s.NewInClusterClient()
```
Comment (awprice) [inline cmd/main.go]: This could be further simplified to just `return k8s.NewInClusterClient()`

File: cmd/main.go [hunk 4]
```diff
@@ -118,18 +116,18 @@ func setupK8SClient(kubeConfigFile *string, leaderElect *bool) kubernetes.Interf
 		}
 		k8sClient, err := k8s.NewOutOfClusterClient(*kubeConfigFile)
```
Comment (awprice) [inline cmd/main.go]: This could be further simplified to just `return k8s.NewOutOfClusterClient(*kubeConfigFile)`

File: pkg/controller/client.go [hunk 5]
```diff
@@ -45,7 +45,7 @@ func NewClient(k8sClient kubernetes.Interface, nodegroups []NodeGroupOptions, st
 
 	synced := k8s.WaitForSync(3, stopCache, podSync, nodeSync)
```
Comment (awprice) [inline pkg/controller/client.go]: Are you able to move the `3` to a const within the method?

**Pull request merged**